app/controllers/step_controller.rb

```diff
@@ -9,12 +9,18 @@ def current_tribunal_case
     @current_tribunal_case ||= TribunalCase.find_by_id(session[:tribunal_case_id])
   end
 
-  def update_and_advance(form_class, opts={})
+  def update_and_advance(attr, form_class, opts={})
     hash = params.fetch(form_class.name.underscore, {})
     @form_object = form_class.new(hash.merge(tribunal_case: current_tribunal_case))
     @next_step = params[:next_step].blank? ? nil : params[:next_step]
 
     if @form_object.save
+      # if we are reusing a step (e.g. income, income2, income3)
+      # we need to rename the 'income' attribute to 'income2' (or whatever) for the
+      # DecisionTree#destination call
+      # i.e. { income: "high" } -> { income2: "high" }
+      hash = opts[:as] ? { opts[:as] => hash[attr] } : hash
```

Comment:
This is to enable re-using a step (erb template) at another place in the tree, but saying "I'm using this form object as: this other thing"

Rails form_for(:foo) will give you a key :foo in the params hash, below which will be all the attributes of foo that the user is submitting. So, if we're using the "bar" form object, but we've specified an as: "foo" key in opts, then give me a hash { "foo" => [whatever was submitted] }. Otherwise, just leave it alone, because it's already { "bar" => [whatever was submitted] }